File: server/internal/auth/delivery/http/dtos.go
package http

type loginRequest struct {
	Email    string `json:"email"`
	Password string `json:"password"`
}

type loginResponse struct {
	AccessToken      string `json:"access_token"`
	RefreshToken     string `json:"refresh_token"`
	AccessExpiresAt  string `json:"access_expires_at"`
	RefreshExpiresAt string `json:"refresh_expires_at"`
}

type registerRequest struct {
	Email    string `json:"email"`
	Password string `json:"password"`
	FullName string `json:"full_name"`
}

type refreshRequest struct {
	RefreshToken string `json:"refresh_token"`
}

type meResponse struct {
	ID       string `json:"id"`
	Email    string `json:"email"`
	FullName string `json:"full_name"`
}

File: client/libs/api/src/lib/features/auth/types/dtos.ts
export interface AuthLoginRequest {
  email: string;
  password: string;
}

export interface AuthRegisterRequest {
  email: string;
  password: string;
  full_name: string;
}

export interface RefreshRequest {
  refresh_token?: string;
}

export interface AuthTokensResponse {
  access_token: string;
  refresh_token?: string;
  access_expires_at: string;
  refresh_expires_at?: string;
}

export interface CreateRoleRequest {
  name: string;
}

export interface AddPermissionRequest {
  permission_id: string;
}

export interface AssignRoleRequest {
  role_id: number;
}
